source/functional-concepts.md

```diff
@@ -0,0 +1,150 @@
+# Functional Programming in Swift: A Primer
+
+Swift is a hybrid language that allows both objective-oriented and functional code to co-exist.
+
+### First-class functions
+
+First-class functions are *values* that can either take other functions as arguments or return them as results. Also refered to as *higher order functions*.
+
+#### Returning Functions
+
+~~~swift
+typealias Prefixer = String -> String
+
+func prefixerFactory(prefix: String) -> Prefixer {
+// func prefix(prefix: String) -> (String -> String)
+
+  func aPrefixer(word: String) -> String {
+    return prefix + word
+  }
+
+  return aPrefixer
+}
+
+let repeat = prefixerFactory("re-")
+repeat("do")
+~~~
+
+*Note*: You can also return an unnamed function, i.e. a closure.
```

Comment:
I will use "anonymous function" and remove closure from the text but mention it verbally. We will not have time to talk about lambdas etc.